pkg/controller/deployment/util/deployment_util.go

```diff
@@ -132,6 +132,18 @@ func GetDeploymentCondition(status extensions.DeploymentStatus, condType extensi
 	return nil
 }
 
+// TODO: remove the duplicate
+// GetDeploymentConditionInternal returns the condition with the provided type.
+func GetDeploymentConditionInternal(status internalextensions.DeploymentStatus, condType internalextensions.DeploymentConditionType) *internalextensions.DeploymentCondition {
```

Comment:
I thought the intention with respect to utilities used by clients was that we are going to have utilities use internal types and clients should run conversions if they wanted to use our utilities, is that correct? If so, why don't you remove the utilities with the external types here and make kubectl convert the object before using the utility?